src/build123d/topology.py

```diff
@@ -3873,111 +4185,99 @@ def face(self) -> Face:
         return compounds[0]
 
-    def __gt__(self, sort_by: Union[Axis, SortBy] = Axis.Z):
+    def __gt__(self, sort_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:  # type: ignore
         """Sort operator >"""
         return self.sort_by(sort_by)
 
-    def __lt__(self, sort_by: Union[Axis, SortBy] = Axis.Z):
+    def __lt__(self, sort_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:  # type: ignore
         """Reverse sort operator <"""
         return self.sort_by(sort_by, reverse=True)
 
-    def __rshift__(self, group_by: Union[Axis, SortBy] = Axis.Z):
+    def __rshift__(self, group_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:
         """Group and select largest group operator >>"""
         return self.group_by(group_by)[-1]
 
-    def __lshift__(self, group_by: Union[Axis, SortBy] = Axis.Z):
+    def __lshift__(self, group_by: Union[Axis, SortBy] = Axis.Z) -> ShapeList[T]:
         """Group and select smallest group operator <<"""
         return self.group_by(group_by)[0]
 
-    def __or__(self, filter_by: Union[Axis, GeomType] = Axis.Z):
+    def __or__(self, filter_by: Union[Axis, GeomType] = Axis.Z) -> ShapeList[T]:
         """Filter by axis or geomtype operator |"""
         return self.filter_by(filter_by)
 
-    def __eq__(self, other: object):
+    def __eq__(self, other: object) -> bool:
         """ShapeLists equality operator =="""
         return (
-            set(self) == set(other) if isinstance(other, ShapeList) else NotImplemented
+            set(self) == set(other) if isinstance(other, ShapeList) else NotImplemented  # type: ignore
         )
 
     # Normally implementing __eq__ is enough, but ShapeList subclasses list,
     # which already implements __ne__, so we need to override it, too
-    def __ne__(self, other: ShapeList):
+    def __ne__(self, other: ShapeList) -> bool:  # type: ignore
         """ShapeLists inequality operator !="""
         return (
             set(self) != set(other) if isinstance(other, ShapeList) else NotImplemented
         )
 
-    def __add__(self, other: ShapeList):
+    def __add__(self, other: ShapeList) -> ShapeList[T]:  # type: ignore
         """Combine two ShapeLists together operator +"""
         return ShapeList(list(self) + list(other))
 
-    def __sub__(self, other: ShapeList) -> ShapeList:
+    def __sub__(self, other: ShapeList) -> ShapeList[T]:
         """Differences between two ShapeLists operator -"""
-        # hash_other = [hash(o) for o in other]
-        # hash_set = {hash(o): o for o in self if hash(o) not in hash_other}
-        # return ShapeList(hash_set.values())
         return ShapeList(set(self) - set(other))
 
-    def __and__(self, other: ShapeList):
+    def __and__(self, other: ShapeList) -> ShapeList[T]:
         """Intersect two ShapeLists operator &"""
         return ShapeList(set(self) & set(other))
 
     @overload
-    def __getitem__(self, key: int) -> T: ...
+    def __getitem__(self, key: SupportsIndex) -> T: ...
```

Comment:
SupportsIndex was required to make mypy happy; otherwise I was redefining `list`.  I had to use a few `# type : ignore` statements in `ShapeList` because mypy doesn't like redefining operators.

Reply:
I got mixed up between [`__index__()`](https://docs.python.org/3/reference/datamodel.html#object.__index__) and `list`'s `.index()`, my bad!